src/django/api/reports.py

```diff
@@ -22,11 +29,80 @@ def monthly_promoted_name_and_address():
                     month = h[1].strftime('%Y-%m')
                     data[month] = data.get(month, 0) + 1
     return [['month', 'facilities promoted'],
-            sorted(data.items(), key=lambda x: x[0])]
+            sort_by_first_column(data.items())]
+
+
+def geocoding_time_without_queue(date_format):
+    temp_data = dict()
+    listitems = FacilityListItem.objects.values_list(
+        'created_at', 'processing_results')
+    for f in listitems:
+        for r in f[1]:
+            if r['action'] == ProcessingAction.GEOCODE:
+                started = datetime.strptime(r['started_at'],
+                                            '%Y-%m-%d %H:%M:%S.%f')
+                finished = datetime.strptime(r['finished_at'],
+                                             '%Y-%m-%d %H:%M:%S.%f')
+                date = started.strftime(date_format)
+                time = finished - started
+                counts = temp_data.get(date, (0, 0))
+                temp_data[date] = (counts[0] + 1,
+                                   counts[1] + time.total_seconds())
+    data = dict()
+    for d, c in temp_data.items():
+        data[d] = c[1] / c[0]
+    return sort_by_first_column(data.items())
+
+
+def monthly_geocoding_time_without_queue():
+    rows = geocoding_time_without_queue('%Y-%m')
+    return [['month', 'average_geocoding_time_in_seconds'], rows]
+
+
+def weekly_geocoding_time_without_queue():
+    rows = geocoding_time_without_queue('%Y-%W')
+    return [['week', 'average_geocoding_time_in_seconds'], rows]
+
+
+def geocoding_time_with_queue(date_format):
+    temp_data = dict()
+    listitems = FacilityListItem.objects.values_list('source__created_at',
```

Comment:
Are we using `source__created_at` rather than `created_at` on the list item itself for a specific reason? They are set at nearly the same time, so I would expect the difference between them to be negligible and using `created_at` would avoid a `JOIN`. 

https://github.com/open-apparel-registry/open-apparel-registry/blob/564bca4adb7f3766e8dfefbbaf2a29293f883f5a/src/django/api/views.py#L2373-L2392

Reply:
I wasn't sure how significant the time difference would be. Since it's minor, I've moved to just using 'created_at'.